### PR TITLE
Adds implementation and demo view controller

### PR DESCRIPTION
**Pull request by shredlocker**
# Why?

We need to have the option to change the state of the bottom sheet from the outside. 

# What?

The BottomSheet can transition from states by setting its `state` property. 
Added demo view controller for testing purposes.
Some refactoring

# Show me

### CODE REVIEW

File: Demo/Demo.swift [hunk 1]
```diff
@@ -42,6 +42,7 @@ public enum ComponentViews: String, CaseIterable {
     case refreshControl
     case horizontalSlide
     case newYearsView
+    case bottomSheetMechanis
```
Comment (vadymmarkov) [inline Demo/Demo.swift]: Should be `bottomSheetMechanics` I guess.

Comment (bstien) [inline Demo/Demo.swift]: Small typo here, should be `bottomSheetMechanics`.

File: Sources/Components/BottomSheet/Controller/Demo/BottomSheetMechanicsDemoViewController.swift [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            compactButton.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -.largeSpacing),
+            compactButton.centerYAnchor.constraint(equalTo: view.centerYAnchor, constant: 0),
+
+            dismissButton.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: .largeSpacing),
+            dismissButton.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -.largeSpacing),
+            dismissButton.bottomAnchor.constraint(equalTo: view.bottomAnchor, constant: -.veryLargeSpacing),
+        ])
+    }
+
+    @objc func expandButtonPressed() {
```
Comment (vadymmarkov) [inline Sources/Components/BottomSheet/Controller/Demo/BottomSheetMechanicsDemoViewController.swift]: You could make these functions `@objc private`

File: Sources/Components/BottomSheet/Controller/Demo/BottomSheetMechanicsDemoViewController.swift [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        view.backgroundColor = .milk
+        view.addSubview(presentButton)
+        NSLayoutConstraint.activate([
+            presentButton.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: .largeSpacing),
+            presentButton.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -.largeSpacing),
+            presentButton.bottomAnchor.constraint(equalTo: view.bottomAnchor, constant: -.veryLargeSpacing),
+        ])
+    }
+
+    @objc func presentButtonPressed() {
```
Comment (vadymmarkov) [inline Sources/Components/BottomSheet/Controller/Demo/BottomSheetMechanicsDemoViewController.swift]: Could also be private.

File: Demo/Demo.swift (hunk 1 above)
Comment (shredlocker) [reply Demo/Demo.swift]: Woops

**Pull request merged**